pkg/utils/kubernetes/leaderelection.go

```diff
@@ -28,14 +28,14 @@ import (
 // ReadLeaderElectionRecord returns the leader election record for a given lock type and a namespace/name combination.
 func ReadLeaderElectionRecord(ctx context.Context, client client.Client, lock, namespace, name string) (*resourcelock.LeaderElectionRecord, error) {
 	switch lock {
-	case resourcelock.EndpointsResourceLock:
+	case resourcelock.EndpointsLeasesResourceLock:
```

Comment:
Instead of using `EndpointsLeasesResourceLock` just don't use any constant (since there is none anymore) and write "endpoints" explicitly here.